core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1792,12 +1792,36 @@ class SparkContext(config: SparkConf) extends Logging {
       }
     }
 
+    def addRemoteJarFile(path: String): String = {
+      val hadoopPath = new Path(path)
+      val scheme = new URI(path).getScheme
+      if (!Array("http", "https", "ftp").contains(scheme)) {
+        try {
+          val fs = hadoopPath.getFileSystem(hadoopConfiguration)
+          if (!fs.exists(hadoopPath)) {
+            throw new FileNotFoundException(s"Jar ${path} not found")
+          }
+          if (fs.isDirectory(hadoopPath)) {
+            throw new IllegalArgumentException(
+              s"Directory ${path} is not allowed for addJar")
+          }
+          path
+        } catch {
+          case NonFatal(e) =>
+            logError(s"Failed to add $path to Spark environment", e)
+            null
```

Comment:
Hm, why null here? if you can't check existence, I'd think you either want to continue anyway with a warning and return path, or throw an exception (or just let the exception throw)?

Reply:
I am confused too, since for local file , it will just continue with a warn, then return a null, the the wrong path won't be add to jar path collection. Seems reasonable too. 
It's reasonable to do like this for dfs file the same.

Reply:
Oh I see, the existing code does that too.